kuksa_databroker/Dockerfile-cli

```diff
@@ -15,27 +15,59 @@
 
 # This builds the databroker-cli
 
-FROM rust:1.65-alpine as builder
+FROM rust:1.65-alpine as base
+RUN rustup component add rustfmt
+RUN apk add git musl-dev python3 protoc ncurses-terminfo-base
+
+FROM base AS builder-amd64
+RUN echo "AMD64 quirks"
+# it seems on AMD build some dependency requires zlib headers
+# also the grpc proto compilation always falls back to
+# some binaries downloaded by cargo that require glibc
+#RUN apk add zlib-dev libc6-compat  libstdc++ gcompat
+
+# We need to specifiy target in build later to enable
+# static linking
+# See https://msfjarvis.dev/posts/building-static-rust-binaries-for-linux/
+# and https://github.com/rust-lang/rust/issues/78210
+ENV BUILDTARGET=x86_64-unknown-linux-musl
```

Comment:
I didn't notice that you didn't specify `--target x86_64-unknown-linux-musl` when building before (i.e. in the databroker Docker file for example).

Wouldn't it make more sense to put all $TARGETARCH specific stuff in their respective `FROM base AS builder-$TARGETARCH`

```suggestion
ENV RUSTFLAGS='-C link-arg=-s'

FROM base AS builder-amd64
RUN echo "AMD64 quirks"
# it seems on AMD build some dependency requires zlib headers
# also the grpc proto compilation always falls back to
# some binaries downloaded by cargo that require glibc
#RUN apk add zlib-dev libc6-compat  libstdc++ gcompat
RUN cargo build  --bin databroker-cli --release --target x86_64-unknown-linux-musl
RUN mv /build/target/x86_64-unknown-linux-musl/release/databroker-cli /dist/
```

and in the final step
```
FROM  scratch

COPY --from=builder /dist/databroker-cli /app/databroker-cli
```


Also not sure why it works at all without adding:
`rustup target add aarch64-unknown-linux-musl`

when cross-compiling..